mastermind.py

```diff
@@ -0,0 +1,125 @@
+import random
+import string
+import matplotlib.pyplot as plt
+
+def generateRandomString(length):
+    characters = string.ascii_letters + string.digits + string.punctuation
+    randomString = ''.join(random.choice(characters) for _ in range(length))
+    return randomString
+
+def fitnessFunction(candidate, target):
+    return sum(1 for a, b in zip(candidate, target) if a == b)
+
+def selectParents(population, fitnessScores, numParents):
+    parents = sorted(zip(population, fitnessScores), key=lambda x: x[1], reverse=True)[:numParents]
+    return [p[0] for p in parents]
+
+def crossover(parent1, parent2):
+    crossoverPoint = random.randint(0, len(parent1) - 1)
+    return parent1[:crossoverPoint] + parent2[crossoverPoint:]
+
+def mutate(candidate, mutationRate):
+    characters = string.ascii_letters + string.digits + string.punctuation
+    candidateList = list(candidate)
+    for i in range(len(candidateList)):
+        if random.random() < mutationRate:
+            candidateList[i] = random.choice(characters)
+    return ''.join(candidateList)
+
+def geneticAlgorithm(testString, populationSize, iterations, elitism, crossoverNum, generation, mutationRate):
```

Comment:
Ez mar jo sok argumentum, ezek egy resze a GA configja, a `teststring` meg igazabol az "instance".  En lehet a tobbit beraknam egy dobozba (egyszerubb megoldas egy dict, szebb egy dataclass mondjuk). Es akkor lehet kifejezobb nevet is adni neki, mert pl hogy mi az a `generation` azt ne mtudom. Meg apro megjegyzes: pythonban inkabb snake_case mint camelCase.